### PR TITLE
feat: resource pool invalidation on startup [DET-9533]

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
On startup, check the `rp_workspace_bindings` table and see if there are any resource pools that don't exist anymore in the config. If they don't exist, mark the `valid` column of those bindings as `false`.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/db/setup.go [hunk 1]
```diff
@@ -49,3 +52,47 @@ func Setup(opts *config.DBConfig) (*PgDB, error) {
...
+
+	ctx := context.Background()
+	tx, err := Bun().BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("error beginning transaction: %t", err)
+	}
+	defer func() {
+		txErr := tx.Rollback()
+		if txErr != nil && txErr != sql.ErrTxDone {
+			log.WithError(txErr).Error("error rolling back transaction in AddExperiment")
```
Comment (jgongd) [inline master/internal/db/setup.go:72]: Just for my understanding, how is ValidateRPWorkspaceBindings() related to AddExperiment?

**Review by jgongd — APPROVED** [review]
LGTM

File: master/internal/db/setup.go (hunk 1 above)
Comment (eecsliu) [reply master/internal/db/setup.go:72]: It's not related to AddExperiment - we just run this on startup to validate resource pools

File: master/internal/core.go [hunk 2]
```diff
@@ -846,6 +846,11 @@ func (m *Master) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+
+	err = db.ValidateRPWorkspaceBindings(m.config.ResourcePools)
+	if err != nil {
+		return err
+	}
 	defer closeWithErrCheck("db", m.db)
```
Comment (stoksc) [inline master/internal/core.go:854]: don't separate the `defer close...` for the db from the creation, so we can make sure we clean up.

File: master/internal/db/setup.go [hunk 3]
```diff
@@ -49,3 +52,47 @@ func Setup(opts *config.DBConfig) (*PgDB, error) {
...
+
+// ValidateRPWorkspaceBindings checks if rp-workspace bindings pertain to valid resource pools.
+// Bindings with resource pools that don't exist have the "valid" column set to false.
+func ValidateRPWorkspaceBindings(pools []config.ResourcePoolConfig) error {
+	var poolNames []string
+	for _, pool := range pools {
+		poolNames = append(poolNames, pool.PoolName)
+	}
+
+	ctx := context.Background()
```
Comment (stoksc) [inline master/internal/db/setup.go:64]: maybe `context.TODO()`, or take it as an arg for now. so it is less forgettable if we give `run` a context.

**Review by stoksc — APPROVED** [review]
lgtm, just some minor comments

File: master/internal/core.go (hunk 2 above)
Comment (stoksc) [reply master/internal/core.go:854]: otherwise it is fine to put it here, right after the db is created.

**Pull request merged**